detection-rules/link_contains_punycode_characters.yml

```
@@ -0,0 +1,15 @@
+name: "Link Contains Punycode Characters"
+description: |
+  The message contains Punycode characters to hide the true URL destination, or contains non-printable ASCII content.
```

Since it's a domain thing:
```suggestion
name: "Link to a Domain with Punycode Characters"
description: |
  The body contains a link to a domain with Punycode characters to hide the true URL destination, or contains non-printable ASCII content.
```